src/MotorPlugin.cc

```diff
@@ -0,0 +1,704 @@
+    }
+    if (!_ecm.Component<gz::sim::components::JointForceCmd>(control.joint))
+    {
+      _ecm.CreateComponent(control.joint, gz::sim::components::JointForceCmd({0.0}));
+    }
+
+    // Subscriber
+    std::string topic = this->impl->topics[i];
+    this->impl->node.Subscribe(
+            topic,
+            &MotorPlugin::Impl::OnPwmMsg, this->impl.get());
+
+    gzdbg << "MotorPlugin subscribing to messages on [" << topic << "]\n";
+  }
+  this->impl->validConfig = true;
+
+
+}
+
+
+//////////////////////////////////////////////////
+void MotorPlugin::ConfigureParameters(
+    gz::transport::parameters::ParametersRegistry &_registry,
+    gz::sim::EntityComponentManager &_ecm)
+{
+  this->impl->registry = &_registry;
+
+  std::string scopedName = gz::sim::scopedName(
+    this->impl->joint.Entity(), _ecm, ".", false);
+  std::string prefix = std::string("MotorPlugin") + scopedName
+    + std::string(".");
+
+  //! @note not using gz::msgs::PID because the message does not support all
+  //!       fields available in gz::math::PID (cmd_max, cmd_min, cmd_offset)
+
+  // Declare parameter proxies
+  this->impl->DeclareParameter(this->impl->pGain,
+      &math::PID::PGain, &math::PID::SetPGain, prefix);
+  this->impl->DeclareParameter(this->impl->iGain,
+      &math::PID::IGain, &math::PID::SetIGain, prefix);
+  this->impl->DeclareParameter(this->impl->dGain,
+      &math::PID::DGain, &math::PID::SetDGain, prefix);
+  this->impl->DeclareParameter(this->impl->iMax,
+      &math::PID::IMax, &math::PID::SetIMax, prefix);
+  this->impl->DeclareParameter(this->impl->iMin,
+      &math::PID::IMin, &math::PID::SetIMin, prefix);
+  this->impl->DeclareParameter(this->impl->cmdMax,
+      &math::PID::CmdMax, &math::PID::SetCmdMax, prefix);
+  this->impl->DeclareParameter(this->impl->cmdMin,
+      &math::PID::CmdMin, &math::PID::SetCmdMin, prefix);
+  this->impl->DeclareParameter(this->impl->cmdOffset,
+      &math::PID::CmdOffset, &math::PID::SetCmdOffset, prefix);
+}
+/////////////////////////////////////////////////
+void MotorPlugin::LoadControlChannels(
+    sdf::ElementPtr _sdf,
+    gz::sim::EntityComponentManager &_ecm)
+{
+  // per control channel 
+  sdf::ElementPtr controlSdf;
+  if (_sdf->HasElement("control"))
+  {
+    controlSdf = _sdf->GetElement("control");
+  }
+  while (controlSdf)
+  {
+    Control control;
+
+    if (controlSdf->HasAttribute("channel"))
+    {
+      control.channel =
+        atoi(controlSdf->GetAttribute("channel")->GetAsString().c_str());
+    }
+    else
+    {
+      // this->impl->channel = this->
+      gzwarn << "[" /*<< this->impl->modelName*/ << "] "
+             <<  "id/channel attribute not specified, use order parsed ["
+             /* << control.channel */ << "].\n";
+    }
+
+    // parameters
+    if (controlSdf->HasElement("jointName"))
+    {
+      control.jointName = controlSdf->Get<std::string>("jointName");
```

Comment:
`gz-sim` uses `lower_case_underscore` for sdf elements. We don't consistently in the `ArduPilotPlugin` due to legacy, but should here. So

```xml
< joint_name> </joint_name>
```
instead of

```xml
< jointName> </jointName>
```

and elsewhere.